.github/workflows/ci.yml

```diff
@@ -46,92 +34,97 @@ jobs:
-            lint: ./scripts/lint-dotnet-worker.sh
-            submodules: true
+            tools: dotnet go
     steps:
-      - name: Print build information
-        run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
       - name: Checkout repo
         uses: actions/checkout@v4
         with:
-          submodules: ${{ matrix.submodules }}
+          submodules: true
       - name: Install mise
         uses: jdx/mise-action@v2.4.4
         with:
           version: ${{ env.MISE_VERSION }}
       - name: Install Tools
-        run: ${{ matrix.install }}
-      - name: Lint ${{ matrix.sdk }}
-        run: ${{ matrix.lint }}
+        run: |
+          go run ./cmd/dev install ${{ matrix.tools }}
+      - name: Add Go bin to PATH
+        run: |
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Check ${{ matrix.sdk }} worker
+        run: |
+          go run ./cmd/dev check ${{ matrix.sdk }}
+      - name: Test ${{ matrix.sdk }} worker
+        run: |
+          go run ./cmd/dev test ${{ matrix.sdk }}
       - name: Check formatting
         run: |
-          git diff --exit-code || (echo "${{ matrix.sdk }} worker has uncommitted formatting changes after linting" && exit 1)
+          git diff --exit-code || (echo "${{ matrix.sdk }} worker has uncommitted formatting changes" && exit 1)
 
   build-worker:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         sdk: [go, java, python, typescript, dotnet]
-        include:
-          - sdk: go
-            install: ./scripts/install-go.sh
-            build: ./scripts/build-go-worker.sh
-            submodules: false
-          - sdk: java
-            install: ./scripts/install-java.sh && ./scripts/install-go.sh
-            build: ./scripts/build-java-worker.sh
-            submodules: false
-          - sdk: python
-            install: ./scripts/install-python.sh && ./scripts/install-go.sh
-            build: ./scripts/build-python-worker.sh
-            submodules: false
-          - sdk: typescript
-            install: ./scripts/install-node.sh && ./scripts/install-go.sh
-            build: ./scripts/build-typescript-worker.sh
-            submodules: true
-          - sdk: dotnet
-            install: ./scripts/install-dotnet.sh && ./scripts/install-go.sh
-            build: ./scripts/build-dotnet-worker.sh
-            submodules: true
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
         with:
-          submodules: ${{ matrix.submodules }}
-      - name: Install mise
-        uses: jdx/mise-action@v2.4.4
-        with:
-          version: ${{ env.MISE_VERSION }}
-      - name: Install Tools
-        run: ${{ matrix.install }}
-      - name: Build ${{ matrix.sdk }} worker
-        run: ${{ matrix.build }}
+          submodules: true
+      - name: Load versions
+        run: |
+          set -a && source versions.env && set +a
```

Comment:
Can we just load this file in the Go scripts? Don't think this needs to be in env var form. Just yaml or even JSON would do.

Reply:
Not blocking the PR, I just think it's a little cleaner to contain everything in the Go scripts and do less bash scripting in CI. It's easier for running everything locally.

Reply:
I see your next comment suggests essentially merging the previous `build` command into the existing Omes `build-worker-image`. That sounds okay to me if we move it into the `dev` command since it doesn't really belong to the Omes CLI imo. Same for `prepare-worker`, actually. Then those two would just be dev commands.